test_scripts/API/SendLocation/001_SendLocation_success.lua

```diff
@@ -0,0 +1,122 @@
+
+-- Steps:
+-- appID requests SendLocation with address, longitudeDegrees, latitudeDegrees, deliveryMode and other parameters
+
+-- Expected:
+
+-- SDL validates parameters of the request
+-- SDL checks if Navi interface is available on HMI
+-- SDL checks if SendLocation is allowed by Policies
+-- SDL checks if deliveryMode is allowed by Policies
+-- SDL transfers the request with allowed parameters to HMI
+-- SDL receives response from HMI
+-- SDL transfers response to mobile app
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSendLocation = require('test_scripts/API/SendLocation/common_send_location')
+
+--[[ Local Variables ]]
+local request_params = {
+    longitudeDegrees = 1.1,
+    latitudeDegrees = 1.1,
+    address = {
+        countryName = "countryName",
+        countryCode = "countryName",
+        postalCode = "postalCode",
+        administrativeArea = "administrativeArea",
+        subAdministrativeArea = "subAdministrativeArea",
+        locality = "locality",
+        subLocality = "subLocality",
+        thoroughfare = "thoroughfare",
+        subThoroughfare = "subThoroughfare"
+    },
+    timeStamp = {
+        millisecond = 0,
+        second = 40,
+        minute = 30,
+        hour = 14,
+        day = 25,
+        month = 5,
+        year = 2017,
+        tz_hour = 5,
+        tz_minute = 30
+    },
+    locationName = "location Name",
+    locationDescription = "location Description",
+    addressLines = 
+    { 
+        "line1",
+        "line2",
+    }, 
+    phoneNumber = "phone Number",
+    deliveryMode = "PROMPT",
+    locationImage = 
+    { 
+        value = "icon.png",
+        imageType = "DYNAMIC",
+    }
+}
+
+--[[ Local Functions ]]
+local function send_location(params, self)
+    local cid = self.mobileSession1:SendRPC("SendLocation", params)
+
+    local deviceID = commonSendLocation.getDeviceMAC()
+    params.locationImage.value = commonSendLocation.getPathToSDL() .. "storage/"
+        .. commonSendLocation.getMobileAppId(1) .. "_" .. deviceID .. "/icon.png"
+
+
+    EXPECT_HMICALL("Navigation.SendLocation", params)
+    :Do(function(_,data)
+        --hmi side: sending Navigation.SendLocation response
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+    :ValidIf (function(_,data)
+        if data.payload.info then
+            print("SDL sent redundant info parameter to mobile App ")
+            return false
+        else 
+            return true
+        end
+    end)
+end
+
+local function put_file(file_name, self)
+    local CorIdPutFile = self.mobileSession1:SendRPC(
+      "PutFile",
+      {syncFileName = file_name, fileType = "GRAPHIC_PNG", persistentFile = false, systemFile = false},
+      "files/icon.png")
```

Comment:
@Itileda Why don't we use an "file_name" argument to get source file? (```files/icon.png```)

Reply:
@dboltovskyi , fixed in 6d9eb2c7cc573896229a5a5a5d4b830581e0e427
parameter file_name removed at all - redundant here.